src/main/java/frc/robot/Commands/IntakeCargo.java

```diff
@@ -15,23 +14,29 @@ public IntakeCargo() {
         IntakeRoller intakeRoller = IntakeRoller.getInstance();
         IntakePlacer intakePlacer = IntakePlacer.getInstance();
         Transfer transfer = Transfer.getInstance();
-        PIDSettings pidSettings = intakePlacer.getPIDSettings();
-        FeedForwardSettings feedForwardSettings = intakePlacer.getFeedForwardSettings();
-        addRequirements(intakeRoller, intakePlacer, transfer);
+        IntakeToTransfer intakeToTransfer = IntakeToTransfer.getInstance();
+        addRequirements(intakeRoller, intakePlacer, transfer, intakeToTransfer);
         addCommands(
-                new MoveGenericSubsystemWithPID(intakePlacer,
-                        () -> IntakePlacer.POTENTIOMETER_RANGE_VALUE,
-                        intakePlacer::getPotentiometerAngle, pidSettings, feedForwardSettings),
-                new MoveGenericSubsystem
-                        (intakeRoller, IntakeRoller.SPEED) {
+                new MoveGenericSubsystem(intakePlacer, IntakePlacer.MIN_SPEED),
+                new ParallelCommandGroup(
+                        new MoveGenericSubsystem(intakeRoller, IntakeRoller.SPEED) {
+                            @Override
+                            public boolean isFinished() {
+                                return intakeToTransfer.getLimit();
+                            }
+                        },
+                        new MoveGenericSubsystem(intakeToTransfer, IntakeToTransfer.SPEED) {
+                            @Override
+                            public boolean isFinished() {
+                                return transfer.isStartPressed();
+                            }
+                        }
+                ),
+                new MoveGenericSubsystem(transfer, Transfer.SPEED) {
                     @Override
                     public boolean isFinished() {
-                        return !transfer.isStartPressed();
+                        return intakeToTransfer.getLimit();
```

Comment:
I might be completely wrong here but why would you want to stop the transfer once the limit that sits at the start of intakeToTransfer gets pushed?

Reply:
This is the parallel command group part. It moves the roller until there is a cargo in the `IntakeToTransfer` and then stops the roller ( all cargos are going to the transfer subsystem, unless there is already a cargo there, then it sits in the `IntakeToTransfer` until we shoot both cargos)

Reply:
The roller is set to work until there is a cargo in the `IntakeToTransfer` subsystem, and from there the cargo either goes to the `Transfer` subsystem or stays in case there is already another cargo there.